src/main/java/org/assertj/core/api/AbstractPathAssert.java

```diff
@@ -0,0 +1,446 @@
+     *
+     * <p>Also note that the fact that a path is absolute does not mean that it
+     * is {@link Path#normalize() normalized}: {@code /foo/..} is absolute, for
+     * instance, but it is not normalized.</p>
+     *
+     * <p>Sample use:</p>
+     *
+     * <pre><code class="java">
+     *     assertThat(somePath).isAbsolute();
+     * </code></pre>
+     *
+     * <p>Sample paths on Unix and Windows which are absolute, or not:</p>
+     *
+     * <ul>
+     *     <li>{@code /foo/bar} on Unix: absolute;</li>
+     *     <li>{@code foo/bar} on Unix: not absolute;</li>
+     *     <li>{@code c:\foo} on Windows: absolute;</li>
+     *     <li>{@code foo\bar} on Windows: not absolute;</li>
+     *     <li>(surprise...) {@code c:foo} on Windows: <strong>not</strong>
+     *     absolute!</li>
+     *     <li>(surprise...) {@code \foo} on Windows: <strong>not</strong>
+     *     absolute!</li>
+     * </ul>
+     *
+     * @return self
+     *
+     * @throws AssertionError path is not an absolute path
+     *
+     * @see Path#isAbsolute()
+     * @see Path#toRealPath(LinkOption...)
+     */
+    public S isAbsolute()
+    {
+        paths.assertIsAbsolute(info, actual);
+        return myself;
+    }
+
+    /**
+     * Assert that a given {@link Path} is normalized
+     *
+     * <p>A path is normalized if it has no redundant components; typically, on
+     * both Unix and Windows, this means that the path has no "self" components
+     * ({@code .}) and that its only parent components ({@code ..}), if any, are
+     * at the beginning of the path.</p>
+     *
+     * Sample usage:
+     *
+     * <pre><code class="java">
+     *     assertThat(path).isNormalized();
+     * </code></pre>
+     *
+     * <p>For instance, {@code a/b} on Unix systems is normalized but {@code
+     * a/./b} is not; on Windows, {@code c:\a\b} is normalized but {@code
+     * c:\a\.\b} is not.</p>
+     *
+     * @return self
+     * @throws AssertionError path is not normalized
+     */
+    public S isNormalized()
+    {
+        paths.assertIsNormalized(info, actual);
+        return myself;
+    }
+
+    /**
+     * Assert whether the tested path is canonical
+     *
+     * <p>For Windows users, this assertion is no different than {@link
+     * #isAbsolute()}. For Unix users, this assertion ensures that the tested
+     * path is the actual file system resource, that is, it is not a {@link
+     * Files#isSymbolicLink(Path) symbolic link} to the actual resource, even if
+     * the path is absolute.</p>
+     *
+     * <p>For instance, if we are in directory {@link /tmp/foo} and the
+     * following shell code is entered:</p>
+     *
+     * <pre><code class="sh">
+     *     # Create an empty file
+     *     $ touch file
+     *     # Create a symbolic link to that file
+     *     $ ln -s file symlink
+     * </code></pre>
+     *
+     * <p>then provided that {@code file} and {@code symlink} are absolute
```

Comment:
Move the examples in the code section with a comment line before each case (not at the end).